docs: storage layout override (#2908)
vyperlang/vyper: docs/compiling-a-contract.rst

    $ vyper -p yourProject yourProject/yourFileName.vy

.. _compiler-storage-layout:

Storage Layout
~~~~~~~~~~~~~~

To display the default storage layout for a contract:

::

    $ vyper -f layout yourFileName.vy

> This outputs a JSON object detailing the default storage location for all state variables as determined by the compiler.

Reviewer: ```suggestion
This outputs a JSON object detailing the locations for all state and immutable variables as determined by the compiler.
```

To override the default storage layout for a contract: 

::

    $ vyper --storage-layout-file storageLayout.json yourFileName.vy

> The input to the `--storage-layout-file` flag must match the format of the storage layout output from the ``vyper -f layout`` command.

Reviewer: oh hm, we changed the format of the layout file

Reviewer: (see updated test cases in https://github.com/vyperlang/vyper/pull/2768)

Reviewer: ```suggestion
The input to the `--storage-layout-file` flag must match the format of the ``.storage_layout`` field from the ``vyper -f layout`` command.
```

> The input to the `--storage-layout-file` flag must match the format of the ``.storage_layout`` field from the ``vyper -f layout`` command.

Reviewer: ```suggestion
The input to the ``--storage-layout-file`` flag must match the format of the ``.storage_layout`` field from the ``vyper -f layout`` command.
```
